examples/pennylane_run_barren_plateaus.py

```diff
@@ -0,0 +1,198 @@
+r"""
+.. _barren_plateaus:
+
+Barren plateaus in quantum neural networks
+==========================================
+*Author: Shahnawaz Ahmed (shahnawaz.ahmed95@gmail.com)*
+
+In classical optimization, it is suggested "that saddle
+points, not local minima, provide a fundamental impediment
+to rapid high-dimensional non-convex optimization".
+(Dauphin et. al, 2014).
+
+The problem of such barren plateaus manifests in a different
+form in variational quantum circuits which are at the heart
+of techniques such as quantum neural networks or approximate
+optimization e.g., QAOA (Quantum Adiabatic Optimization Algorithm).
+
+While starting from a parameterized
+random quantum circuit seems like a good unbiased choice if
+we do not know the problem structure, McClean et. al., 2018,
+show that
+
+*"for a wide class of reasonable parameterized quantum
+circuits, the probability that the gradient along any
+reasonable direction is non-zero to some fixed precision
+is exponentially small as a function of the number
+of qubits."*
+
+Thus, randomly selected quantum circuits might not be the best
+option to choose while implementing variational quantum
+algorithms.
+
+
+.. figure:: ../../examples/figures/barren_plateaus/surface.png
+   :scale: 52%
+   :alt: surface
+
+In this tutorial, we will show how randomized quantum circuits 
+face the problem of barren plateaus using Pennylane. We will
+partly reproduce some of the findings in McClean et. al., 2018
+with just a few lines of code.
+
+
+Initialization strategy to tackle barren plateaus
+-------------------------------------------------
+
+In (Grant et. al., 2019), the authors present one strategy to
+tackle the barren plateau problem in randomized quantum circuits:
+
+*"The technique involves randomly selecting some of the initial
+parameter values, then choosing the remaining values so that
+the final circuit is a sequence of shallow unitary blocks that
+each evaluates to the identity. Initializing in this way limits
+the effective depth of the circuits used to calculate the first
+parameter update so that they cannot be stuck in a barren plateau
+at the start of training."*
+"""
+
+##############################################################################
+# Exploring the barren plateau problem with Pennylane
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+import pennylane as qml
+from pennylane import numpy as np
+import matplotlib.pyplot as plt
```

Comment:
in terms of the structural flow, the heading "Exploring the barren plateau problem with PennyLane" is at the same level as the below "Create a randomized variational circuit", but only contains import statements. To make it clearer, it might be worth adding a short ReST comment above ("First, we import PennyLane, NumPy, and Matplotlib"), or making the headings below subheadings.

Reply:
Done